includes/fserver.entity.inc

```diff
@@ -172,31 +175,113 @@ function release_uri($entity) {
 }
 
 /**
- * Options list callback for the supported_majors property
+ * Entity Property Callbacks
  */
-function _fserver_project_property_supported_majors_options() {
-  // @todo make this intelligent by looking at the attached releases
-  // @todo
-  return array(
-    '5.x' => t('5.x'),
-    '6.x' => t('6.x'),
-    '7.x' => t('7.x'),
-    '8.x' => t('8.x'),
-    '9.x' => t('9.x'),
-  );
+
+/**
+ * Entity property getter callback for default_major
+ */
+function fserver__property_serialized_get($project, array $options, $name) {
+  $return = NULL;
+  $name = isset($info['schema field']) ? $info['schema field'] : $name;
+  if ((is_array($project) || (is_object($project) && $project instanceof ArrayAccess)) && isset($project[$name])) {
+    if (is_array($project[$name])) {
+      $return = $project[$name];
+    }
+    else {
+      $return = unserialize($project[$name]);
+    }
+  }
+  // @todo get rid of the i18n stuff
+  elseif (is_object($project) && isset($project->$name)) {
+
+    if (is_array($project->$name)) {
+      $return = $project->$name;
+    }
+    else {
+      $return = unserialize($project->$name);
+    }
+  }
+
+  return $return;
 }
 
 /**
- * Options list callback for the recommended_majors property
+ * Entity property getter callback for default_major
  */
-function _fserver_project_property_recommended_majors_options() {
-  // @todo make this intelligent by looking at the attached releases
-  return array(
-    1 => 1,
-    2 => 2,
-    3 => 3,
-    4 => 4,
-  );
+function fserver__property_serialized_set(&$data, $name, $value, $langcode, $type, $info) {
+  // Serialize the value, the rest stays the same.
+  $value = serialize($value);
+  $name = isset($info['schema field']) ? $info['schema field'] : $name;
+  if (is_array($data) || (is_object($data) && $data instanceof ArrayAccess)) {
+    $data[$name] = $value;
+  }
+  elseif (is_object($data)) {
+    $data->$name = $value;
+  }
+}
+
+/**
+ * Entity property validation callback for default_major
+ *
+ * @see fserver_project_default_major_property_info().
+ */
+function fserver__project_property_default_major_validate($value) {
+  if (is_array($value)) {
+    foreach ($value as $key => $v) {
+      if (array_key_exists('api', $v) and array_key_exists('default_major', $v)) {
```

Comment:
Nope. Entity Property Info for the List type only allows integer keys. Then we get into using a struct by that doesn't work either cause it requires fixed key / value pair definitions.